### DISCUSSION TITLE
Sharing trpc between a t3 app and a Node app

### DISCUSSION THREAD

**Discussion by galnir** [trpc/trpc#2472] [Q&A] [answered]
What is the best way to define a tRPC router in a t3 app and then use it in the t3 app and in a Node app, for this matter a Discord bot? The 2 apps are each in its own folder but both folders in the same git repo.
I tried importing AppRouter in the Node app but during the build it throws TS errors originating from files in the Next app, stuff like prisma errors and "fetch is not a function".
I also tried doing something similar to the zART example but am getting more errors.

**Comment by sachinraja** [comment, marked as answer]
I suggest extracting the trpc router to a separate package, although you shouldn't be getting errors here. We can't help more without a reproduction.

**Reply by galnir** [reply]
> 

Alright so my question is, is it best practice to extract the router to a separate package and then import it in both the Discord bot app and the Next app, and in each I initialize a tRPC client, passing it the AppRouter?

**Reply by sachinraja** [reply]
Yes, that's what I would do.
